templates.js

```diff
@@ -1,8 +1,8 @@
 'use strict';
-const TEMPLATE_REGEX = /(?:\\(u[a-f\d]{4}|x[a-f\d]{2}|.))|(?:\{(~)?(\w+(?:\([^)]*\))?(?:\.\w+(?:\([^)]*\))?)*)(?:[ \t]|(?=\r?\n)))|(\})|((?:.|[\r\n\f])+?)/gi;
+const TEMPLATE_REGEX = /(?:\\(u(?:[a-f\d]{4}|\{[a-f\d]{1,5}\})|x[a-f\d]{2}|.))|(?:\{(~)?(\w+(?:\([^)]*\))?(?:\.\w+(?:\([^)]*\))?)*)(?:[ \t]|(?=\r?\n)))|(\})|((?:.|[\r\n\f])+?)/gi;
```

Comment:
The Unicode standard [allows](https://www.unicode.org/charts/PDF/U10FF80.pdf) six-character code points. Would it make sense to allow this?

Additionally, in the spec, [unicode escape sequences](https://read262.netlify.com/ecmascript-language-lexical-grammar/ecmascript-language-lexical-grammar-literals#prod-UnicodeEscapeSequence) with braces like this allow [a hex value ≤ 0x10FFFF](https://read262.netlify.com/ecmascript-language-lexical-grammar/ecmascript-language-lexical-grammar-literals#prod-CodePoint).

Reply:
Yep, good catch. I was testing with a bad value.